hphp/runtime/vm/jit/relocation-arm.cpp

```diff
@@ -231,9 +237,16 @@ bool relocateSmashable(Env& env, Instruction* src, Instruction* dest,
  * destCount, srcCount and rewrites are updated to reflect when an
  * instruction(s) is rewritten to a different instruction sequence.
  */
-bool relocatePCRelative(Env& env, Instruction* src, Instruction* dest,
+bool relocatePCRelative(Env& env, TCA srcAddr, TCA destAddr,
                         size_t& srcCount, size_t& destCount) {
-  auto const destAddr = reinterpret_cast<TCA>(dest);
+
+  auto const srcFrom = Instruction::Cast(srcAddr);
+  auto const destFrom = Instruction::Cast(destAddr);
+
+  auto const srcAddrActual = env.srcBlock.toDestAddress(srcAddr);
+  auto const destAddrActual = env.destBlock.toDestAddress(destAddr);
+  auto const src = Instruction::Cast(srcAddrActual);
+  auto const dest = Instruction::Cast(destAddrActual);
```

Comment:
Is updating Instruction to track a virtual and real address out of the question? We did that for DecodedInstruction on the x64 side and I think it made stuff like this a lot less verbose.